dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/ZenodoHelper.java

```diff
@@ -149,19 +209,53 @@ private static void setMetadataKeywords(DepositMetadata depositMetadata, Workflo
      * @param workflow    workflow for which DOI is registered
      */
     private static void setMetadataRelatedIdentifiers(DepositMetadata depositMetadata, String dockstoreUrl,
-            Workflow workflow) {
-        // Get the aliases for this workflow and add them to the deposit
-        // The alias must be a format supported by Zenodo such as
-        // DOI, Handle, ARK...URNs and URLs
-        // See http://developers.zenodo.org/#representation 'related identifiers'
-        List<RelatedIdentifier> aliasList = workflow.getAliases().keySet().stream()
-                .map(alias -> {
-                    RelatedIdentifier relatedIdentifier = new RelatedIdentifier();
-                    String aliasUrl = dockstoreUrl + "/aliases/workflows/" + alias;
-                    relatedIdentifier.setIdentifier(aliasUrl);
-                    relatedIdentifier.setRelation(RelatedIdentifier.RelationEnum.ISIDENTICALTO);
-                    return relatedIdentifier;
-                }).collect(Collectors.toList());
+            Workflow workflow, WorkflowVersion workflowVersion, String alias,
+            EntryVersionHelper entryVersionHelper, DockstoreWebserviceConfiguration webserviceConfiguration) {
+
+        List<RelatedIdentifier> aliasList = new ArrayList<>();
+
+        // Add the workflow version alias as a related identifier on Zenodo
+        String aliasUrl = dockstoreUrl + "/aliases/workflows/" + alias;
+        RelatedIdentifier aliasRelatedIdentifier = new RelatedIdentifier();
+        aliasRelatedIdentifier.setIdentifier(aliasUrl);
+        aliasRelatedIdentifier.setRelation(RelatedIdentifier.RelationEnum.ISIDENTICALTO);
+        aliasList.add(aliasRelatedIdentifier);
+
+        // Add the UI2 link to the workflow to Zenodo as a related identifier
+        // E.g https://dockstore.org/workflows/aliases/10.5281/zenodo.2630727
+        String sourceControl = workflow.getSourceControl().toString(); // e.g. github.com
+        String organization = workflow.getOrganization(); // e.g. DataBiosphere
+        String repository = workflow.getRepository(); // e.g. topmed-workflows
+        String workflowName = workflow.getWorkflowName(); // e.g. UM_variant_caller_wdl
+        String versionOfWorkflow = workflowVersion.getName();
+
+        String sourceControlPathPrefix = sourceControl + "/" + organization + "/" + repository;
+        // Sometimes the workflow name returned is null so skip it if that is the case
+        // this enables a correct UI2 link and TRS URL to be constructed
+        // E.g. https://dockstore.org/workflows/github.com/DataBiosphere/topmed-workflows/UM_variant_caller_wdl:1.32.0
+        String sourceControlPath = StringUtils.isEmpty(workflowName) ? sourceControlPathPrefix : sourceControlPathPrefix + "/" + workflowName;
+        String workflowVersionURL = dockstoreUrl + "/workflows/"  + sourceControlPath + ":" + versionOfWorkflow;
```

Comment:
Is this different from `MetadataResource.createWorkflowURL()`?
We use that to generate URLs for the sitemap and RSS feed

Reply:
Now using MetadataResource.createWorkflowURL()